internal/kafka/test/integration/cluster_metrics_test.go

```diff
@@ -0,0 +1,94 @@
+package integration
+
+import (
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/constants"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/kafkas/types"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test/common"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test/mocks/kasfleetshardsync"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/test/mocks"
+	. "github.com/onsi/gomega"
+)
+
+func TestClusterCapacityUsedMetric(t *testing.T) {
+	// create a mock ocm api server, keep all endpoints as defaults
+	// see the mocks package for more information on the configurable mock server
+	ocmServer := mocks.NewMockConfigurableServerBuilder().Build()
+	defer ocmServer.Close()
+
+	h, _, teardown := test.NewKafkaHelperWithHooks(t, ocmServer, nil)
+	defer teardown()
+
+	mockKasFleetshardSyncBuilder := kasfleetshardsync.NewMockKasFleetshardSyncBuilder(h, t)
+	mockKasfFleetshardSync := mockKasFleetshardSyncBuilder.Build()
+	mockKasfFleetshardSync.Start()
+	defer mockKasfFleetshardSync.Stop()
+
+	// setup pre-requisites to performing requests
+	db := test.TestServices.DBFactory.New()
+
+	// create a dummy cluster and assign a kafka to it
+	cluster := &api.Cluster{
+		ClusterID:             "some-cluster-id",
+		MultiAZ:               true,
+		Region:                "us-east-1",
+		CloudProvider:         "aws",
+		Status:                api.ClusterReady,
+		IdentityProviderID:    "some-id",
+		ClusterDNS:            "some-cluster-dns",
+		ProviderType:          api.ClusterProviderStandalone,
+		SupportedInstanceType: "standard",
+		ClientID:              "some-client-id",
+		ClientSecret:          "some-client-secret",
+	}
+
+	kafka := &dbapi.KafkaRequest{
+		Meta: api.Meta{
+			ID: api.NewID(),
+		},
+		MultiAZ:                 false,
+		Owner:                   "some-owner",
+		Region:                  cluster.Region,
+		CloudProvider:           cluster.CloudProvider,
+		Name:                    "test-kafka",
+		OrganisationId:          orgId,
+		Status:                  constants.KafkaRequestStatusReady.String(),
+		ClusterID:               cluster.ClusterID,
+		ActualKafkaVersion:      "2.6.0",
+		DesiredKafkaVersion:     "2.6.0",
+		ActualStrimziVersion:    "v.23.0",
+		DesiredStrimziVersion:   "v0.23.0",
+		InstanceType:            types.STANDARD.String(),
+		ReauthenticationEnabled: false,
+		SizeId:                  "x1",
+	}
```

Comment:
Can we build this using the kafka mocks? https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/test/mocks/kafkas/kafkas.go

Reply:
same comment as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1101#discussion_r900283720